harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -247,10 +235,6 @@ def __init__(
         self.global_batch_size = global_batch_size
         self.profiling_enabled = profiling_enabled
 
-        if self.searcher_unit == core.Unit.RECORDS:
-            if self.global_batch_size is None:
-                raise ValueError("global_batch_size required for searcher unit RECORDS.")
-
```

Comment:
It occurs to me, seeing this case no longer supported here, that you need to add a `max_length` to the pytorch code in `exec/harness.py`, and you need to probably have this exact check over there.

Reply:
no... nvm, I think you have it correct.

I think you need the shim to be Trainer, not in harness.py, to cover all the cases.